lockrenewal.go

```diff
@@ -0,0 +1,77 @@
+package servicebus
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-amqp-common-go/log"
+	"github.com/Azure/azure-amqp-common-go/rpc"
+	otlogger "github.com/opentracing/opentracing-go/log"
+	"time"
+
+	"pack.ag/amqp"
+)
+
+const (
+	serviceBuslockRenewalOperationName = "com.microsoft:renew-lock"
+)
+
+//RenewLocks renews the locks on messages provided
+func (e *entity) RenewLocks(ctx context.Context, messages []*Message) error {
```

Comment:
Looking at the PR history, I see that @devigned gave what amounts to contradicting feedback. I guess my $0.02 on this are:

- Between automatic and manual message lock renewal, I'm not sure which way ought to be the 99% case. Automatic will protect people who have taken too long to process for legitimate reasons, like message processing just being computationally expensive. However, it potentially hides bugs that have caused deadlock. At the end of the day, I don't have a strong opinion here, we just need to document our decision.
- I believe the `receiver` should be the one doing the automatic message lock renewal. Passing in a `receiverOption` that configures it to do so. It then parlays that into calling `Message.RenewLock` periodically as appropriate.

Reply:
Yeah it's a tough one to know either way. I'd usually look to take prior art and consistency between the SB SDK's as a guide - in the existing .NET SDK I *believe* that autorenewal is the default behavior and I've never seen people handling this manually. 

Because of this I'd structured the `RenewLock` method to sit on the entity and accepted batches of messages to allow `Receiver` to use this call efficiently when the AutoLocking functionality is added. By efficiently I'm thinking each receiver keep a track of all active messages and renew the batch with a single RPC round trip vs having 100 open messages and having to make 100 RPC calls - I can see a common usecase having a large number of messages active at one time. This also feeds into the locking comment too, as renewal occurs in batch infrequently a mutex per entity seems like a fair tradeoff. 

Then the method is also available on the entity for those who want to use it manually with autorenewal turned off. 

What do you think?

Reply:
You've convinced me. :)

I think down the road, I may add a method off of Message so that you can either RenewLock directly off the message, or continue to use the batch method that you're adding here.